provider/cmd/leaseStatus.go

```diff
@@ -23,37 +22,45 @@ func leaseStatusCmd() *cobra.Command {
 		},
 	}
 
-	mcli.AddBidIDFlags(cmd.Flags())
-	mcli.MarkReqBidIDFlags(cmd)
+	addLeaseFlags(cmd)
 
 	return cmd
 }
 
 func doLeaseStatus(cmd *cobra.Command) error {
-	cctx := client.GetClientContextFromCmd(cmd)
+	cctx, err := client.ReadTxCommandFlags(client.GetClientContextFromCmd(cmd), cmd.Flags())
+	if err != nil {
+		return err
+	}
 
-	addr, err := mcli.ProviderFromFlagsWithoutCtx(cmd.Flags())
+	prov, err := providerFromFlags(cmd.Flags())
 	if err != nil {
 		return err
 	}
 
-	pclient := pmodule.AppModuleBasic{}.GetQueryClient(cctx)
-	res, err := pclient.Provider(context.Background(), &ptypes.QueryProviderRequest{Owner: addr.String()})
+	dseq, gseq, oseq, err := parseLeaseFromFlags(cmd.Flags())
```

Comment:
Can you fold this into `x/market/client/cli/flags.go` and `x/deployment/cli/flags.go`?

I am worried about constructing a `LeaseID` below that doesn't have `Owner` populated.